Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,183 @@
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.Actions.Components;
+using Content.Shared.Audio;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.MobState.EntitySystems;
+using Content.Shared.Popups;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+
+namespace Content.Server.Guardian
+{
+    public class GuardianSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<GuardianCreatorComponent, UseInHandEvent>(OnGuardianCreated);
+            SubscribeLocalEvent<GuardianCreatorComponent, ExaminedEvent>(OnActivatorExamined);
+            SubscribeLocalEvent<GuardianComponent, MoveEvent>(OnGuardianMove);
+            SubscribeLocalEvent<GuardianHostComponent, MoveEvent>(OnGuardianHostMove);
+            SubscribeLocalEvent<GuardianComponent, DamageChangedEvent>(OnGuardianDamaged);
+            SubscribeLocalEvent<GuardianHostComponent, MobStateChangedEvent>(OnHostStateChange);
+        }
+
+        /// <summary>
+        /// Triggers when the host recives damage which puts the host in either critical or killed state
+        /// </summary>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (EntityManager.GetEntity(component.Hostedguardian) != null)
+            {
+                if (args.State.IsCritical())
+                {
+                    var guard = EntityManager.GetEntity(component.Hostedguardian);
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-host-critical-warn"), guard.Transform.Coordinates, Filter.Entities(guard.Uid));
+                    SoundSystem.Play(Filter.Local(), "/Audio/Effects/guardian_warn.ogg", uid);
```

Comment:
Sounds should be fields on the class as SoundSpecifier which means linter can actually catch it.

Reply:
Any good example of this?